datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -162,9 +162,14 @@ public DatastreamTaskImpl(List<Datastream> datastreams, String id, List<Integer>
    * @param partitionsV2 new partitions for this task
    */
   public DatastreamTaskImpl(DatastreamTaskImpl predecessor, Collection<String> partitionsV2) {
+    _dependencies = new ArrayList<>();
     if (!predecessor.isLocked()) {
-      throw new DatastreamTransientException("task " + predecessor.getDatastreamTaskName() + " is not locked, "
-          + "the previous assignment has not been picked up");
+      if (!predecessor._partitionsV2.isEmpty()) {
```

Comment:
Combine these two conditions in a single if statement? That would make code a bit more readable.